lib/python/bind_units.cpp

```diff
@@ -5,16 +5,77 @@
 #include "scipp/core/dtype.h"
 #include "scipp/core/tag_util.h"
 #include "scipp/units/unit.h"
-#include "scipp/variable/variable.h"
 
-#include "dtype.h"
-#include "numpy.h"
 #include "pybind11.h"
 #include "unit.h"
 
 using namespace scipp;
 namespace py = pybind11;
 
+constexpr int UNIT_DICT_VERSION = 1;
+
+py::dict to_dict(const units::Unit &unit) {
+  py::dict dict;
+  dict["__version__"] = UNIT_DICT_VERSION;
+
+  const auto &base_units = unit.underlying().base_units();
+  dict["meter"] = base_units.meter();
+  dict["kilogram"] = base_units.kg();
+  dict["second"] = base_units.second();
+  dict["ampere"] = base_units.ampere();
+  dict["kelvin"] = base_units.kelvin();
+  dict["mole"] = base_units.mole();
+  dict["candela"] = base_units.candela();
+  dict["currency"] = base_units.currency();
+  dict["count"] = base_units.count();
+  dict["radian"] = base_units.radian();
+  // Returning ints instead of bools because
+  // - The constructor of unit_data takes unsigned int as arguments.
+  // - h5py saves bools as enum types which waste disk space.
+  dict["per_unit"] = static_cast<unsigned int>(base_units.is_per_unit());
+  dict["i_flag"] = static_cast<unsigned int>(base_units.has_i_flag());
+  dict["e_flag"] = static_cast<unsigned int>(base_units.has_e_flag());
+  dict["equation"] = static_cast<unsigned int>(base_units.is_equation());
```

Comment:
I don't think we are using these? Error codes are translated into exceptions in our wrapper. We should avoid storing information that ties as to a specific units library, I think.

Reply:
We are potentially using them via user defined units:
```.py
print(sc.Unit('CXUN[123]').to_dict())
{'__version__': 1, 'meter': 7, 'kilogram': -2, 'ampere': -32, 'kelvin': 3, 'mole': -8, 'currency': -2, 'i_flag': 1, 'e_flag': 1, 'multiplier': 1.0}
```

Reply:
This would mean dropping https://github.com/scipp/scipp/pull/2808

Reply:
Then I must have misunderstood how that feature works. I thought it simply links into the string formatting, not changing how actual units are stored?

Reply:
It hooks into formatting, yes. It can make the formatter prefer certain units over others, e.g. angstrom over nm. But the reason it would break our serialisation is that it can introduce new units. And as I understood @g5t, the use case for this were custom units (`CXUN[...]`).

We can still follow through with #2808 without supporting custom units. But I think it would limit its usefulness. Unless I am missing something and people like to define their own names for combinations of SI units.

Reply:
> Unless I am missing something and people like to define their own names for combinations of SI units.
That was my understanding, e.g., to define things in units of lattice spacing.